components/skus/browser/rs/lib/src/sdk/credentials/fetch.rs

```diff
@@ -63,17 +96,120 @@ where
             return Err(InternalError::OrderUnpaid.into());
         }
 
-        let mut csprng = OsRng;
 
         for item in order.items {
             match item.credential_type {
+                CredentialType::TimeLimitedV2 => {
+                    let mut num_creds: usize = 0;
+                    if let Some(ref metadata) = order.metadata {
+                        let num_intervals = metadata.num_intervals.unwrap_or(0);
```

Comment:
should we raise an error in the case that `num_intervals` / `num_per_interval` is not present?